src/Generator/CodeGenerator.php

```diff
@@ -468,4 +607,45 @@ function ($a, $b) use ($imports) {
 
         return $unique_imports;
     }
+
+    /**
+     * Ensures the enity class complies to the "standard" for holding parameters.
+     *
+     * @param string $entity_class
+     */
+    private function validateEnumEntity(string $entity_class)
+    {
+        $reflector = new \ReflectionClass($entity_class);
+
+        if (! $reflector->hasMethod('__construct')) {
+            throw new \LogicException(sprintf(
+                'Entity class "%s" must implement __construct($owning_entity, $name) in order to be Enum-compatible.',
```

Comment:
You _could_ require an interface with a constructor for this, perhaps even make a trait. The current check doesn't enforce the signature, some for the other method checks in here.